### PR TITLE
Add restart UI

### LINKED ISSUE #539: Implement restart UI
When using e10s you might need to restart when you install a restartless add-on. #507 will add support for detecting this but we need to implement the UI from https://github.com/mozilla/addons/issues/64. The mocks show a "Restart" link but I don't think we can restart Firefox.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #539

The `UNINSTALLING` restart case only occurs if someone clicks remove from about:addons#Extensions. Handling the user clicking `undo` is covered by #562.

Both kinds of restart look like this:

add-ons_manager

### CODE REVIEW

File: src/disco/css/Addon.scss [hunk 1]
```diff
@@ -120,4 +92,40 @@ $addon-padding: 20px;
...
+    display: flex;
+    flex-direction: row;
+    left: 0;
+    padding: 15px 20px;
+    position: absolute;
+    right: 0;
+    top: 0;
+
+    &.error {
+      background: #c33c32 url('../img/warning.svg') no-repeat calc(100% - 40px) 50%;
```
Comment (mstriemer) [inline src/disco/css/Addon.scss]: Will webpack inline these or do they need to be `require`d?

Comment (muffinresearch) [inline src/disco/css/Addon.scss]: Yeah it should inline it since it's tiny.

File: src/disco/css/Addon.scss [hunk 2]
```diff
@@ -120,4 +92,40 @@ $addon-padding: 20px;
...
+      z-index: 20;
```
Comment (mstriemer) [inline src/disco/css/Addon.scss]: Why is the error higher than needs restart? So you see an error if it is in `ERROR` and `needsRestart`?

Comment (muffinresearch) [inline src/disco/css/Addon.scss]: Yeah, it's very unlikely to be the case that both would be displayed at the same time, but I figured that an error should trump the restart notification.

**Comment by mstriemer** [comment]
Do we see the `needsRestart` state on a reload? Should you be able to close the UNINSTALLING notice? You could presumably uninstall by accident and want to install again.

r+wc

**Comment by muffinresearch** [comment]
> Do we see the needsRestart state on a reload?

No, since we don't get that hint from the introspection. I'm not sure if that would be an issue or not.

>  Should you be able to close the UNINSTALLING notice? You could presumably uninstall by accident and want to install again.

It's possible we could have an undo instead of close, since the unstalling state is still pending. It'll certainly be worth looking at that separately. I'll file a new issue to cover that. Maybe 'undo' could work for all restart cases?

**Comment by mstriemer** [comment]
If it is `needsRestart` then the state should be `INSTALLING` or `UNINSTALLING`, right? I think we could just do the opposite in the case of an undo, good call.

**Pull request merged**